File: Lista_Logica_Programacao/Python/ex10.py
# Exercício 10 - Gerenciador de Tarefas

# programa que permite adicionar e listar tarefas em um arquivo de texto

while True:
    # mostra o menu de opções
    print("\n=== Gerenciador de Tarefas ===")
    print("1 - Adicionar tarefa")
    print("2 - Listar tarefas")
    print("3 - Sair")

    escolha = input("Escolha uma opção (1-3): ")

    if escolha == "3":
        print("Saindo...")
        break

    # opção 1: adicionar tarefa
    if escolha == "1":
        tarefa = input("Digite a tarefa: ")
        # open com "a" abre o arquivo em modo append (adicionar)
        arquivo = open("tarefas.txt", "a")
        # escrevo a tarefa no arquivo
        arquivo.write(tarefa + "\n")
        # sempre fechar o arquivo depois de usar
        arquivo.close()
        print("Tarefa adicionada com sucesso!")

    # opção 2: listar tarefas
    elif escolha == "2":
        try:
            # open com "r" abre em modo read (leitura)
            arquivo = open("tarefas.txt", "r")
            # read() lê todo o conteúdo do arquivo
            conteudo = arquivo.read()
            arquivo.close()

            # split("\n") divide o texto por quebras de linha
            tarefas = conteudo.split("\n")

            print("\n=== Tarefas ===")
            # enumerate() numera automaticamente cada item da lista
            for numero, tarefa in enumerate(tarefas, 1):
                # só mostra se a tarefa não estiver vazia
                if tarefa:
                    print(f"{numero} - {tarefa}")

        except FileNotFoundError:
            # erro caso o arquivo não exista ainda
            print("Nenhuma tarefa ainda. Adicione uma primeiro!")

    else:
        print("Opção inválida, tente novamente.")

# Anotações:
# - open("arquivo.txt", "a") abre o arquivo em modo append para adicionar linhas.
# - open("arquivo.txt", "r") abre em modo read para ler o conteúdo.
# - sempre usar arquivo.close() depois de usar o arquivo.
# - write() escreve texto no arquivo.
# - read() lê todo o conteúdo como uma string.
# - split("\n") divide a string em uma lista usando a quebra de linha como separador.
# - enumerate() retorna índice e valor simultaneamente, começando do 1 neste caso.
# - try/except trata erros, como quando o arquivo não existe ainda.
# - if tarefa: verifica se a tarefa não está vazia antes de exibir.

File: Lista_Logica_Programacao/Java/Ex10.java
// Exercício 10 - Gerenciador de Tarefas

import java.util.Scanner;
import java.io.FileWriter;
import java.io.FileReader;
import java.io.BufferedReader;
import java.io.IOException;

public class Ex10 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (true) {
            System.out.println("\n=== Gerenciador de Tarefas ===");
            System.out.println("1 - Adicionar tarefa");
            System.out.println("2 - Listar tarefas");
            System.out.println("3 - Sair");

            System.out.print("Escolha uma opção (1-3): ");
            int escolha = sc.nextInt();
            sc.nextLine();  // consome a quebra de linha

            if (escolha == 3) {
                System.out.println("Saindo...");
                break;
            }

            // opção 1: adicionar tarefa
            if (escolha == 1) {
                System.out.print("Digite a tarefa: ");
                String tarefa = sc.nextLine();

                try {
                    // FileWriter com true abre em modo append (adicionar)
                    FileWriter arquivo = new FileWriter("tarefas.txt", true);
                    // escrevo a tarefa no arquivo
                    arquivo.write(tarefa + "\n");
                    // sempre fechar o arquivo depois de usar
                    arquivo.close();
                    System.out.println("Tarefa adicionada com sucesso!");

                } catch (IOException e) {
                    System.out.println("Erro ao salvar a tarefa!");
                }
            }

            // opção 2: listar tarefas
            else if (escolha == 2) {
                try {
                    // FileReader para ler o arquivo
                    FileReader fileReader = new FileReader("tarefas.txt");
                    // BufferedReader facilita ler linha por linha
                    BufferedReader leitor = new BufferedReader(fileReader);

                    System.out.println("\n=== Tarefas ===");
                    String linha;
                    int numero = 1;

                    // readLine() lê uma linha por vez até chegar ao fim (null)
                    while ((linha = leitor.readLine()) != null) {
                        System.out.println(numero + " - " + linha);
                        numero++;
                    }

                    leitor.close();
                    fileReader.close();

                } catch (IOException e) {
                    System.out.println("Nenhuma tarefa ainda. Adicione uma primeiro!");
                }
            }

            else {
                System.out.println("Opção inválida, tente novamente.");
            }
        }

        sc.close();
    }
}
/*
Anotações:

- FileWriter é usado para escrever em arquivo.
- new FileWriter("tarefas.txt", true) abre em modo append (true = adicionar).
- FileReader é usado para ler arquivo.
- BufferedReader facilita leitura linha por linha.
- write() escreve texto no arquivo.
- readLine() lê uma linha do arquivo, retorna null quando chega ao fim.
- close() sempre fechar o arquivo depois de usar.
- try/catch trata erros como arquivo não encontrado.
- IOException é a exceção lançada em operações de arquivo.
- a variável numero conta as tarefas manualmente neste caso.
*/
